crates/revm/src/db/ethersdb.rs

```diff
@@ -31,29 +33,51 @@ impl<M: Middleware> EthersDB<M> {
             Some(Self {
                 client,
                 block_number,
-                handle,
+                rt,
             })
         } else {
             let mut instance = Self {
-                client: client.clone(),
+                client,
                 block_number: None,
-                handle,
+                rt,
             };
             instance.block_number = Some(BlockId::from(
-                instance.block_on(client.get_block_number()).ok()?,
+                instance.block_on(instance.client.get_block_number()).ok()?,
             ));
             Some(instance)
         }
     }
 
+    // Create a new EthersDB instance, with a provider and a block (None for latest) and a runtime.
+    //
+    // Refer to [tokio::runtime::Builder] how to create a runtime if you are in synchronous world.
+    // If you are already using something like [tokio::main], call EthersDB::new instead.
+    pub fn with_runtime(
+        client: Arc<M>,
+        block_number: Option<BlockId>,
+        runtime: Runtime,
```

Comment:
so the caller is responsible for creating a new runtime?
in which case they could also pass in the handle?

Reply:
Make sense, we can add a new function and document the behavior.